File: cmms-frontend/src/api/mailSettings.ts
import { api } from './client';

export interface MailSettings {
  enabled: boolean;
  mailType: 'SMTP' | 'SENDGRID';
  fromEmail: string | null;
  fromName: string | null;
  sendgridApiKeySet: boolean;
  smtpHost: string | null;
  smtpPort: number | null;
  smtpUsername: string | null;
  smtpPasswordSet: boolean;
  smtpTrustAllCertificates: boolean;
}

export interface UpdateMailSettingsPayload {
  enabled: boolean;
  mailType: 'SMTP' | 'SENDGRID';
  fromEmail?: string;
  fromName?: string;
  sendgridApiKey?: string;
  smtpHost?: string;
  smtpPort?: number;
  smtpUsername?: string;
  smtpPassword?: string;
  smtpTrustAllCertificates?: boolean;
}

export const mailSettingsApi = {
  get: () => api.get<MailSettings>('/settings/mail'),
  update: (data: UpdateMailSettingsPayload) => api.put<MailSettings>('/settings/mail', data),
  sendTest: (toEmail: string) => api.post<void>('/settings/mail/test', { toEmail }),
};

File: cmms-api/src/main/java/com/tuempresa/cmms/model/entity/MailSettings.java
package com.tuempresa.cmms.model.entity;

import com.tuempresa.cmms.tenant.BaseTenantEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

/**
 * Configuracion de correo por organizacion, editable desde Ajustes sin
 * reiniciar el servidor. Si una organizacion no tiene registro aqui, se
 * usan los valores por defecto de application.yml (variables de entorno).
 */
@Entity
@Table(name = "mail_settings")
@Getter
@Setter
public class MailSettings extends BaseTenantEntity {

    private Boolean enabled = false;

    @Column(nullable = false)
    private String mailType = "SMTP"; // SMTP | SENDGRID

    private String fromEmail;
    private String fromName;

    private String sendgridApiKey;

    private String smtpHost;
    private Integer smtpPort;
    private String smtpUsername;
    private String smtpPassword;

    // Para servidores SMTP internos con certificados autofirmados (comun en
    // redes corporativas). Desactiva la validacion estricta de la cadena de
    // certificacion -- solo recomendable para servidores dentro de tu propia red.
    private Boolean smtpTrustAllCertificates = false;
}
